src/libs/antares/study/parts/thermal/cluster.cpp

```diff
@@ -421,6 +456,66 @@ void Data::ThermalCluster::calculationOfSpinning()
     }
 }
 
+void Data::ThermalCluster::calculationOfMarketBidPerHourAndMarginalCostPerHour()
+{
+    if (costgeneration == Data::setManually || !prepro)
+        costGenManualCalculationOfMarketBidAndMarginalCostPerHour();
```

Comment:
Move `fuelcost` and `co2cost` from `prepro` to `series`. `prepro` is meant to contain data related to TS generation